spec/scope_resolver_spec.rb

```diff
@@ -0,0 +1,80 @@
+require "spec_helper"
+
+module Pundit
+  describe ScopeResolver do
+    subject { described_class.new(user, resource_scope) }
+    let(:user) { double('pundit_user') }
+
+    describe '#resolve' do
+      context 'policy class cannot be found' do
+        let(:resource_scope) { nil }
+
+        it { expect(subject.resolve).to be nil }
+      end
+
+      context 'policy class with not a defined Scope class' do
+        let(:resource_scope) { None }
+
+        before {
+          expect("#{resource_scope}Policy".safe_constantize).to be_present
+          expect("#{resource_scope}Policy::Scope".safe_constantize).to_not be_present
```

Comment:
Are these testing our gem code, or are they testing that our test setup is correct? Could something go wrong if we deleted these `before` blocks?